pkg/auth/azure_auth.go

```diff
@@ -100,9 +100,20 @@ func GetServicePrincipalToken(config *AzureAuthConfig, env *azure.Environment, r
 		}
 		if len(config.UserAssignedIdentityID) > 0 {
 			klog.V(4).Info("azure: using User Assigned MSI ID to retrieve access token")
-			return adal.NewServicePrincipalTokenFromMSIWithUserAssignedID(msiEndpoint,
-				resource,
-				config.UserAssignedIdentityID)
+			resourceId, err := azure.ParseResourceID(config.UserAssignedIdentityID)
+			if err == nil &&
+				strings.EqualFold(resourceId.Provider, "Microsoft.ManagedIdentity") &&
+				strings.EqualFold(resourceId.ResourceType, "userAssignedIdentities") {
+				klog.V(4).Info("azure: User Assigned MSI ID is resource ID")
+				return adal.NewServicePrincipalTokenFromMSIWithIdentityResourceID(msiEndpoint,
+					resource,
+					config.UserAssignedIdentityID)
+			} else {
+				klog.V(4).Info("azure: User Assigned MSI ID is client ID")
```

Comment:
should be error here instead of info

Reply:
Resource ID paring error doesn't necessarily mean a config error because it could be a client id (GUID). I will include the error message in the info log for better diagnostics.